components/timezone/src/provider/names.rs

```diff
@@ -0,0 +1,91 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! 🚧 \[Unstable\] Property names-related data for this component
+//!
+//! <div class="stab unstable">
+//! 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+//! including in SemVer minor releases. While the serde representation of data structs is guaranteed
+//! to be stable, their Rust representation might not be. Use with caution.
+//! </div>
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use crate::TimeZoneBcp47Id;
+use core::str;
+use icu_provider::prelude::*;
+use zerotrie::ZeroTrie;
+use zerovec::{VarZeroVec, ZeroVec};
+
+/// A mapping from IANA time zone identifiers to BCP-47 time zone identifiers.
+///
+/// Multiple IANA time zone IDs can map to the same BCP-47 time zone ID.
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. While the serde representation of data structs is guaranteed
+/// to be stable, their Rust representation might not be. Use with caution.
+/// </div>
+#[derive(Debug, Clone, PartialEq)]
+#[icu_provider::data_struct(marker(
+    IanaToBcp47MapV1Marker,
+    "time_zone/iana_to_bcp47@1",
+    singleton
+))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_timezone::provider::names),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct IanaToBcp47MapV1<'data> {
+    /// A map from IANA time zone identifiers to indexes of BCP-47 time zone identifiers.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub map: ZeroTrie<ZeroVec<'data, u8>>,
```

Comment:
`ZeroTrie` is still experimental, I'd like to see it graduate before we use it in non-experimental code. IIUC we still haven't decided on its hashing algorithm, so the encoding itself is not stable.

Reply:
I can make another PR to move it to utils if you like. There are no open questions about its hashing algorithm.